frontend/src/test/java/com/google/tripmeout/frontend/servlet/PlaceParentServletTest.java

```diff
@@ -0,0 +1,222 @@
+package com.google.tripmeout.frontend.servlet;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.PlaceDetails;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import com.google.tripmeout.frontend.serialization.GsonModelSerializationModule;
+import com.google.tripmeout.frontend.servlets.PlaceParentServlet;
+import com.google.tripmeout.frontend.storage.InMemoryPlaceVisitStorage;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+public class PlaceParentServletTest {
+  @Mock HttpServletRequest request;
+  @Mock PlaceDetailsRequest detailsRequest;
+  private Gson gson;
+
+  private final GeoApiContext CONTEXT =
+      new GeoApiContext.Builder().apiKey("AIzaSyBbXCXC2uWv3baNmirLtqUYbFsFwCXqLV8").build();
+
+  private static final PlaceVisitModel UK = PlaceVisitModel.builder()
+                                                .setName("UK Trip")
+                                                .setTripId("abc123")
+                                                .setPlaceId("ChIJ3S-JXmauEmsRUcIaWtf4MzE")
```

Comment:
I realize now that this may be a real place ID taken from the places API. If so, it's generally best practice to not check in real (possibly sensitive) object keys with code or test data. Luckily, it looks like [place ids](https://developers.google.com/places/web-service/place-id#save-id) are exempt from caching policies and can likely be safely saved and reused; despite that, it's good to maintain the practice of not storing real data (especially from external services) in code like this.